fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java

```diff
@@ -36,13 +39,23 @@
     private final List<FHIRRegistryResourceProvider> providers;
 
     private FHIRRegistry() {
-        providers = loadProviders();
+        providers = new CopyOnWriteArrayList<>(loadProviders());
     }
 
     public static FHIRRegistry getInstance() {
         return INSTANCE;
     }
 
+    /**
+     * Register a provider
+     *
+     * @param provider
+     *     the provider to register
+     */
+    public void register(FHIRRegistryResourceProvider provider) {
```

Comment:
Maybe explain that this shouldn't be called by consumers and that they should use the ServiceLoader to make their static RegistryResourceProviders available instead

Reply:
Added Javadoc